[CS2113-T18-1] ChessMaster

ken-ruster opened the pull request:
ChessMasterCLI is a command-line interface (CLI) chess game designed to make learning and training accessible for beginners while offering an engaging experience for all skill levels. This sleek and user-friendly chess simulator provides a platform for novice players to build their skills and understanding of the game.

maanyos commented on src/main/java/chessmaster/user/Player.java:
```diff
@@ -0,0 +1,171 @@
+
+import java.util.ArrayList;
+
+public abstract class Player {
```
good choice of abstract class for player!

maanyos commented on src/main/java/chessmaster/user/Player.java:
```diff
@@ -0,0 +1,171 @@
+            row = 0;
+        }
+
+        for (int tempRow = row; row < tempRow + 2; row++) {
```
would make more sense if tempRow was changed instead of row

maanyos commented on src/main/java/chessmaster/user/Player.java:
```diff
@@ -0,0 +1,171 @@
+     * @return null if the user inputs "abort", an empty Move if an error occurred
+     *         during parsing, otherwise return the requested Move object.
+     */
+    public Move getNextMove(ChessBoard board) throws InvalidMoveException, NullPieceException, ParseCoordinateException{
+        String input = TextUI.getUserInput(); // Get user input
+        try {
+            Parser parser = new Parser();
+            Command command = parser.parseCommand(input, board);
+            if (command instanceof AbortCommand) {
+                return null;
+            }
+            return command.getMove();
+        
+        } catch (ParseCoordinateException | NullPieceException e) {
+            TextUI.printErrorMessage(e);
+        }
+        return new Move();
+    }
```
is the catch clause intended to throw the same exception again after printing error message?

maanyos commented on src/main/java/chessmaster/parser/Parser.java:
```diff
@@ -0,0 +1,146 @@
+        int colour = promoteFrom.getColour();
+        Coordinate position = promoteFrom.getPosition();
+
+        switch (promoteTo.toLowerCase()){
```
good versatility

maanyos commented on src/main/java/chessmaster/pieces/ChessPiece.java:
```diff
@@ -0,0 +1,175 @@
+
+import java.util.ArrayList;
+
+public abstract class ChessPiece {
```
good use of abstract class for chess piece!

maanyos commented on src/main/java/chessmaster/storage/Storage.java:
```diff
@@ -0,0 +1,104 @@
+    public static void saveBoard(ChessBoard board) throws SaveBoardException {
+
+        try (FileWriter fileWriter = new FileWriter(filePath)){
+            for (int row = 0; row < 8; row++) {
```
try do define constants as final variables on top

maanyos reviewed (commented):
Good organisation of packages, use of exceptions, and documentations!

maanyos commented on src/main/java/chessmaster/ChessMaster.java:
```diff
@@ -0,0 +1,89 @@
+        game.run();
+    }
+
+    public static void main(String[] args) {
+        new ChessMaster().run();
+    }
```
good abstraction of main

maanyos commented on src/main/java/chessmaster/commands/MoveCommand.java:
```diff
@@ -0,0 +1,45 @@
+    }
+
+    public Move getMove() throws InvalidMoveException {
+        if (move == null) {
+            throw new InvalidMoveException();
+        }
```
might be more informative if you add a message argument to the exception or use a different exception
can also consider using assertion if this is intended to ensure code correctness

maanyos commented on src/main/java/chessmaster/game/ChessBoard.java:
```diff
@@ -0,0 +1,378 @@
+import chessmaster.pieces.Pawn;
+import chessmaster.ui.TextUI;
+
+public class ChessBoard {
```
this class is doing a lot!
consider extracting parts that deal with pieces and moves

maanyos commented on src/main/java/chessmaster/game/MiniMax.java:
```diff
@@ -0,0 +1,124 @@
+        //gets all the moves for the current player
+        ChessBoard board = tuple.getBoard();
+        Move[] moves = board.getAllMoves(color);
+        assert moves.length > 0 : "No moves available for " + color + " at depth " + depth;
```
can moves.length = 0 ever happen during use?

maanyos commented on src/main/java/chessmaster/game/Move.java:
```diff
@@ -0,0 +1,96 @@
+        this.to = to;
+        this.piece = piece;
+
+        assert from != null && to != null : "Coordinates in Move should not be null!";
+        assert piece != null && !piece.isEmptyPiece() : "Chess piece in Move should not be null or empty!";
```
can these happen during run time?

maanyos commented on src/main/java/chessmaster/pieces/ChessPiece.java:
```diff
@@ -0,0 +1,275 @@
+    protected int points = 0;
+
+    //initialise empty boardweights of 0 for parent class to be used for the AI
+    private int[][] boardWeight = 
+        {{0,0,0,0,0,0,0,0},
+        {0,0,0,0,0,0,0,0},
+        {0,0,0,0,0,0,0,0},
+        {0,0,0,0,0,0,0,0},
+        {0,0,0,0,0,0,0,0},
+        {0,0,0,0,0,0,0,0},
+        {0,0,0,0,0,0,0,0},
+        {0,0,0,0,0,0,0,0}};
```
no need to initialise here right

maanyos commented on src/main/java/chessmaster/pieces/ChessPiece.java:
```diff
@@ -0,0 +1,275 @@
+     */
+    public Coordinate[] getFlattenedCoordinates(ChessBoard board) {
+        Coordinate[][] availableCoordinates = getAvailableCoordinates(board);
+        ArrayList<Coordinate> flattenedCoordinates = new ArrayList<>();
+
+        for (Coordinate[] direction : availableCoordinates) {
+            for (Coordinate possibleCoord : direction) {
+                if (this.isMoveValid(possibleCoord, board)) {
+                    flattenedCoordinates.add(possibleCoord);
+                }
+            }
```
consider refactoring this part if its used in many methods

maanyos reviewed (commented):
Good that you've added help messages
Good use of documentation and comments in some parts to explain code
Try to keep the way of printing messages consistent, some use const final strings, others put the string directly in the methods
Keep the way of printing output consistent also, some classes are directly calling system.out.print

azfarulmatin commented on docs/DeveloperGuide.md:
```diff
@@ -23,16 +137,29 @@
 |Version| As a ... | I want to ... | So that I can ...|
```
is it supposed to be formatted this way?

azfarulmatin commented on docs/DeveloperGuide.md:
```diff
@@ -6,7 +6,121 @@
+**API:**
+
+Below is a class diagram representing the Storage class.
+The Storage component is responsible for handling the storage and retrieval of chess game state.
```
Should the StorageClass diagram be shown at the next line instead?

ziyi105 commented on docs/diagrams/ChessMasterSequence.puml:
```diff
@@ -0,0 +1,56 @@
+
+    chessMaster -> ui : shouldStartNewGame()
+    activate ui
+    user -> ui : "yes/no"
```
Is this supposed to be the condition? Because its meaning kinda overlaps with Previous Board Exist, so maybe you should just use one of them

ziyi105 commented on docs/diagrams/ParseCommandSequence.puml:
```diff
@@ -0,0 +1,48 @@
+    Command -> ChessBoard ++: showAvailableCoordinates
+    ChessBoard --
+    Command -> ChessBoard ++: getAvailableCoordinatesString
+    ChessBoard --> Command --: String
```
Perhaps using coordinateString: String is clearer?
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/82555990/c6f354b6-f0f8-4ab2-b76d-aa268088f175)

azfarulmatin commented on docs/DeveloperGuide.md:
```diff
@@ -6,7 +6,121 @@
+
+![](StorageSequence.png)
+
+* Creates the necessary parent directories for the file and the file itself if they don't exist 
```
It would be helpful to provide more detailed explanations alongside the diagrams to offer readers a better understanding of creating parent directories, saving the ChessBoard state, resetting the game, and loading the state.

azfarulmatin commented on docs/DeveloperGuide.md:
```diff
@@ -6,7 +6,121 @@
+In order to handle user input into the program during the game, the `Parser` class was implemented.
+Below is a sequence diagram describing the process of handling user input passed from `Game`:
+
+![](images/ParseCommandSequence.png)
```
Great job on the diagram and explanation of how the Parser class works! The visual representation along with the detailed step-by-step breakdown makes it easy to understand the process.

rohitcube commented on docs/images/ChessMasterSequence.png:
Really detailed Sequence diagram. Explains the interactions between the entities simply and clearly.

ziyi105 commented on docs/StorageSequence.puml:
```diff
@@ -0,0 +1,117 @@
+
+
+create Storage
+User -> Storage++ : new Storage(): String
```
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/82555990/f0de8914-63e3-4780-8182-a9ec49844eca)
I am not sure how your code is implemented, but it seems a bit strange to me to have user calling the command to create a new Storage() object. Did you mean user will input a command which is received by a class and the class triggers the creation of storage object? If yes, what is the class that triggers the creation of storage object? Perhaps in your uml you should have an arrow pointing from the class to storage class instead?

DextheChik3n commented on docs/StorageSequence.png:
This sequence diagram seems too big, perhaps break down into the different method calls

ziyi105 commented on docs/StorageSequence.puml:
```diff
@@ -0,0 +1,117 @@
+Storage -> Storage++ : createChessMasterFile()
+Storage --> Storage--
+create FileWriter
+Storage -> FileWriter++ : new FileWriter(): File
```
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/82555990/0e4ea6bb-065c-4bd7-a603-d9f1334bd388)
Is there supposed to be a lifeline for FileWriter?

rohitcube commented on docs/images/GameSequence.png:
Good sequence diagram to explain the interactions between entities for hasGameEnded and isMoveCommand.

DextheChik3n commented on docs/images/ParseCommandSequence.png:
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/35828587/e152baba-f1c7-4ae5-8bed-325fd72430db)
is there a way to prevent the red cross from overlapping with the String return arrow? I feel that this could be misinterpreted

DextheChik3n commented on docs/StorageSequence.png:
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/35828587/52d2899d-a8dc-46f8-b7dc-6008f4bd2430)
is there supposed to be a method call here?

DextheChik3n commented on docs/StorageSequence.png:
![image](https://github.com/nus-cs2113-AY2324S1/tp/assets/35828587/3bce562b-ef84-44d0-bbae-388873de12d6)
Perhaps colon was forgotten before the class name 😅

DextheChik3n reviewed (commented):
Added review comments for DG

ziyi105 reviewed (commented):
Overall clean and neat!

TongZhengHong committed:
Merge pull request #115 from onx001/StartBug

Start bug fix

onx001 committed:
Fix minimax aggro

onx001 committed:
Update MiniMax.java

TongZhengHong committed:
Merge pull request #116 from onx001/Check

Add `isChecked` and `isCheckMated` features

TongZhengHong committed:
Merge pull request #149 from onx001/MinimaxAggro

Fix minimax aggro

TongZhengHong committed:
Update no available moves ui

TongZhengHong committed:
Fix empty payload in ShowMovesCommand

TongZhengHong committed:
Fix wrong color when player loses

TongZhengHong committed:
Convert code blocks from markdown to html

TongZhengHong committed:
Fix link to legend command

onx001 committed:
Merge pull request #152 from TongZhengHong/fix-ug-formatting

Fix User Guide formatting

onx001 committed:
Merge pull request #151 from TongZhengHong/fix-wrong-losing-color

Fix wrong color when player loses

onx001 committed:
Merge pull request #150 from TongZhengHong/fix-moves-command

Fix empty payload in moves command

TongZhengHong committed:
Add InvalidCommand class

TongZhengHong committed:
Update NullPieceException message

TongZhengHong committed:
Check for empty move

ken-ruster committed:
Merge pull request #15 from AY2324S1-CS2113-T18-1/master

Update

TriciaBK committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2113-T18-1/tp

ken-ruster committed:
Add function to save move history

ken-ruster committed:
Update documentation

TriciaBK committed:
Add expected output to UserGuide

TongZhengHong committed:
Check hasNextLine in getUserInput

TongZhengHong committed:
Add CPU thinking message

TongZhengHong committed:
Add shouldIgnoreEmpty flag in getUserInput

TongZhengHong committed:
Ignore empty inputs only for Game commands

TongZhengHong committed:
Save board state at end of loading new game

TongZhengHong committed:
Show difficulty in UI message when continue previous game

onx001 committed:
Merge pull request #154 from TongZhengHong/add-invalid-command

Add invalid command

onx001 committed:
Merge pull request #153 from TongZhengHong/fix-force-quit-error

Fix minor UI and UX bugs in `ChessMaster` class

TriciaBK committed:
Add more details for special rules

TriciaBK committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2113-T18-1/tp into documentation

TriciaBK committed:
Fix Checkstyle

TriciaBK committed:
Update castling rules

ken-ruster committed:
Add promotion as move type

onx001 committed:
Add En Passant and new Check function

Added en passant support in gameplay.

To-do: Save en passant state

Added checking function with parallel move validity.

To-do: Incorporate into show moves and minimax algorithms

onx001 committed:
Implement Check in MiniMax

onx001 committed:
FixCheckBug

onx001 committed:
Add Castling Check

onx001 committed:
Implement checkmate as endgame condition

ken-ruster committed:
Add checks for validity of saved board

ken-ruster committed:
Merge pull request #16 from AY2324S1-CS2113-T18-1/master

Update

onx001 committed:
Rename function and integrate show moves

ken-ruster committed:
Resolve Gradle checks

Commit:
Add basic history functionality

Commit:
Add tests for HistoryCommand

Commit:
Update UG

Commit:
Update DG

Commit:
Fix checkstyle issues

TriciaBK committed:
Update formatting

TongZhengHong committed:
Merge pull request #155 from TriciaBK/documentation

Add details to User Guide

TongZhengHong committed:
Merge pull request #156 from onx001/EnPassant

Add En Passant and new Check function

TongZhengHong committed:
Merge pull request #157 from onx001/CheckAI

Check ai

onx001 committed:
FIx minimax validity

ken-ruster committed:
Merge branch 'master' into saveMoves

ken-ruster committed:
Resolve checkStyle

Commit:
Update end-to-end tests to allow CI to pass

Commit:
Create MoveFactory

Commit:
Update HistoryCommand implementation

Commit:
Update test to match output

TongZhengHong committed:
Merge pull request #160 from onx001/CheckAI

FIx minimax validity

Commit:
Add 'capture' messages for UI and history output

ken-ruster committed:
Resolve comments

ken-ruster committed:
Refactor validity checking

ken-ruster committed:
Merge branch 'master' into saveMoves

onx001 committed:
Fix En Passant Bug

onx001 committed:
Merge branch 'master' into CheckAI

TongZhengHong committed:
Merge pull request #158 from ken-ruster/saveMoves

Add function to save moves, check validity of saved board

TriciaBK committed:
Add restart new game function if game ends or is aborted

TriciaBK committed:
Add restart command to restart board midgame

TriciaBK committed:
Fix bugs with abort and restart

TriciaBK committed:
Update help command

TriciaBK committed:
Fix checkstyle

Commit:
Add basic stepback implementation

Commit:
Add logic to reverse capturing

Commit:
Update UG and help menu

Commit:
Fix help menu alignment

antrikshdhand committed:
Merge branch 'master' into branch-History

antrikshdhand committed:
Update HistoryTest.java

antrikshdhand committed:
Update HistoryTest.java

antrikshdhand committed:
Update HistoryTest.java to fit CheckStyle

antrikshdhand committed:
Merge pull request #159 from antrikshdhand/branch-History

Add basic history functionality & end-to-end testing capabilities

Commit:
Update Unicode in testing file

antrikshdhand committed:
Merge branch 'AY2324S1-CS2113-T18-1:master' into branch-History

TriciaBK committed:
Renamed abort to exit

Commit:
Update branch with latest code

Commit:
Fix checkstyle issues

TriciaBK committed:
refactor code

TriciaBK committed:
Fix checkstyle

TriciaBK committed:
Fix typo

TriciaBK committed:
Fix merge conflicts

Commit:
Update README.md

onx001 committed:
Merge branch 'master' into CheckAI

onx001 committed:
Merge pull request #163 from onx001/CheckAI

Fix En Passant Bug

onx001 committed:
Merge branch 'master' into branch-StepBack

onx001 committed:
Merge pull request #165 from antrikshdhand/branch-StepBack

Implement `stepback` command

TriciaBK committed:
Update command class

onx001 committed:
Move Rename Fix

onx001 committed:
Reimplement check

TongZhengHong committed:
Merge branch 'master' into restart-game

TriciaBK committed:
Change getpiece function

TongZhengHong committed:
Merge pull request #168 from onx001/move-rename

Move Rename Fix

TongZhengHong committed:
Update to getPseudoLegalCoord and getLegalCoord in ChessPiece

TongZhengHong committed:
Update ChessBoard getLegalMoves and getPseudoLegalMoves

TongZhengHong committed:
Change pseudoLegalCoord to return 1D flattened array

TongZhengHong committed:
Fix warnings in storage

TongZhengHong committed:
Merge branch 'master' into branch-refactor-chessboard

TongZhengHong committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2113-T18-1/tp into branch-refactor-chessboard

TongZhengHong committed:
Merge branch 'branch-refactor-chessboard' of https://github.com/TongZhengHong/tp into branch-refactor-chessboard

TongZhengHong committed:
Fix infinite loop by chnaging to getPseudoLegalCoords

TriciaBK committed:
Resolve conflicts

TriciaBK committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2113-T18-1/tp into restart-game

TriciaBK committed:
Update documentation

TriciaBK committed:
Resolve conflicts

TriciaBK committed:
Merge pull request #164 from TriciaBK/restart-game

Add restart game ability

onx001 committed:
Merge pull request #169 from TongZhengHong/branch-refactor-chessboard

Refactor `ChessPiece` processing functions

TriciaBK committed:
Add exit flag

TriciaBK committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2113-T18-1/tp into exit-game

TriciaBK committed:
Add exit function mid load game

TriciaBK committed:
Fix checkstyle

ken-ruster committed:
Merge pull request #17 from AY2324S1-CS2113-T18-1/master

Update

ken-ruster committed:
Update class diagrams

ken-ruster committed:
Update sequence diagrams

ken-ruster committed:
Update architecture diagram

Commit:
Merge branch 'master' of github.com:AY2324S1-CS2113-T18-1/tp into branch-readme

ken-ruster committed:
Add PPP for ken-ruster

ken-ruster committed:
Update JavaDoc documentation

ken-ruster committed:
Fix checkStyle errors

ken-ruster committed:
Merge pull request #170 from TriciaBK/exit-game

Exit game

onx001 committed:
Create onx001.md

onx001 committed:
Update onx001.md

Commit:
Merge branch 'master' into branch-readme

Commit:
Update README.md

Commit:
Merge branch 'master' into branch-history

Commit:
Update small TextUI details

Commit:
Update UG for exit command

ken-ruster committed:
Merge pull request #171 from antrikshdhand/branch-readme

Update README.md

ken-ruster committed:
Merge pull request #172 from ken-ruster/doc-update

Update documentation and fix comments by TA

Commit:
Update stepback and history commands with castling logic

TongZhengHong committed:
Merge pull request #173 from onx001/ppp

Create onx001.md

TongZhengHong committed:
Add draft PPP contributions

TongZhengHong committed:
Rename TongZhengHong.md to tongzhenghong.md

TongZhengHong committed:
Merge pull request #174 from TongZhengHong/branch-zhenghong-ppp

Add draft PPP contributions

TriciaBK committed:
Add PPP draft

TongZhengHong committed:
Merge pull request #175 from TriciaBK/tricia

Add PPP draft

Commit:
Add CastleMove class as a subclass of Move

Commit:
Update MoveCommand message to reflect castling

Commit:
Merge branch 'master' into branch-History

Commit:
Fix checkstyle errors

Commit:
Upload PPP

antrikshdhand committed:
Merge pull request #178 from antrikshdhand/branch-readme

Upload PPP

onx001 committed:
EnPassant load fix

onx001 committed:
Merge pull request #176 from antrikshdhand/branch-History

Update stepback and history commands with castling logic

ken-ruster committed:
Merge pull request #179 from onx001/PassantBug

Fix bug when loading in board with en passant state

onx001 committed:
Use SLAP to simplify minimax

Commit:
Merge stash with upstream

Commit:
Add logic to reverse promotion

Commit:
Refactor en passant moves into its own class

Commit:
Remove redundant code

Commit:
Extract executeMove() code into individual methods

Commit:
Fix checkstyle

onx001 committed:
Merge pull request #180 from antrikshdhand/branch-History

Refactor en passant and add logic for the reversal of promotions

onx001 committed:
Add Draw Condition

onx001 committed:
Cleanup

onx001 committed:
Merge branch 'master' into 2.1-Updates

Commit:
Implement 'captured' command

Commit:
Update UG

Commit:
Fix formatting on help menu

onx001 committed:
Docs Update

Commit:
Update README.md with new command

antrikshdhand committed:
Merge pull request #181 from onx001/2.1-Updates

2.1 updates

onx001 committed:
Update PPP

onx001 committed:
Update README.md

ken-ruster committed:
Merge pull request #18 from AY2324S1-CS2113-T18-1/master

Update

ken-ruster committed:
Fix stepback on loaded game

ken-ruster committed:
Add tests for parsePromote

ken-ruster committed:
Add test for showMoveCommand

onx001 committed:
Merge pull request #183 from onx001/2.1-Updates

Update PPP

onx001 committed:
Merge branch 'master' into branch-Pieces

onx001 committed:
Merge pull request #184 from ken-ruster/stepback-bugfix

Fix bug with stepback not working on loaded game

onx001 committed:
Merge pull request #182 from antrikshdhand/branch-Pieces

Implement 'captured' command

onx001 committed:
Update Documentation

Added sequence diagram for move execution and updated general docs.

TriciaBK committed:
Merge pull request #186 from onx001/docs2

Update Documentation

TriciaBK committed:
Update puml in Developer Guide

TriciaBK committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2113-T18-1/tp into tricia

TriciaBK committed:
Update Developer Guide

TriciaBK committed:
Update Developer Guide

ken-ruster committed:
Merge pull request #19 from AY2324S1-CS2113-T18-1/master

Update

TriciaBK committed:
Update user guide

TriciaBK committed:
fix checkstyle'

TriciaBK committed:
Update UG and DG

ken-ruster committed:
Add test for showMoveCommand

ken-ruster committed:
Add tests for ExecuteSavedMoves

ken-ruster committed:
Correct tests for ShowMovesCommand

ken-ruster committed:
Correct checkStyle

ken-ruster committed:
Add test for executeMoveArray

ken-ruster committed:
Correct checkStyle

ken-ruster committed:
Correct CI tests

ken-ruster committed:
Update PPP

ken-ruster committed:
Update reposense

ken-ruster committed:
Fix CI checks

TongZhengHong committed:
Merge pull request #187 from TriciaBK/tricia

Update Documentations

TongZhengHong committed:
Merge pull request #188 from ken-ruster/add-tests

Add tests

TongZhengHong committed:
Test fix ug formatting of chessboard

TriciaBK committed:
Merge pull request #189 from TongZhengHong/branch-fix-ug-formatting

Test fix ug formatting of chessboard

TongZhengHong committed:
Change ``` to html tags for code blocks

TriciaBK committed:
Merge pull request #190 from TongZhengHong/branch-fix-ug-2

Change ``` to html tags for code blocks

TongZhengHong committed:
Change to table for code block

TongZhengHong committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2113-T18-1/tp into branch-fix-ug-3

TriciaBK committed:
Update documentation and class diagram

TriciaBK committed:
Update PPP

TriciaBK committed:
Merge pull request #191 from TongZhengHong/branch-fix-ug-3

Change to table for code block

Commit:
Update PPP

antrikshdhand committed:
Merge pull request #192 from TriciaBK/tricia

Update DG and PPP

ken-ruster committed:
Merge pull request #193 from antrikshdhand/branch-FinalChanges

Update PPP

ken-ruster committed:
Merge pull request #21 from AY2324S1-CS2113-T18-1/master

Update

ken-ruster committed:
Update PPP

ken-ruster committed:
Update PPP

onx001 committed:
Update docs

Cleared up move segment in developer's  guide and adopted CONTRIBUTORS page.

TongZhengHong committed:
Show storage file format in UG

TongZhengHong committed:
Remove unnecessary var

TongZhengHong committed:
Fix UI message for choose player color

TongZhengHong committed:
Check empty spaces and delimiter in Storage class

TongZhengHong committed:
Merge pull request #195 from onx001/docs3

Update docs

onx001 committed:
Merge pull request #194 from ken-ruster/add-tests

Update PPP

onx001 committed:
Merge pull request #196 from TongZhengHong/branch-rebust-storage

Ignore empty spaces in each `nextLine` method in `Storage` class

onx001 committed:
Update onx001.md

ken-ruster committed:
Fix bug with current turn in save

TriciaBK committed:
Add PPP link

ken-ruster committed:
Merge branch 'master' into add-tests

TongZhengHong committed:
Merge pull request #197 from onx001/ppp1

Update onx001.md

TongZhengHong committed:
Merge pull request #198 from ken-ruster/add-tests

Fix bug with current turn in save

TriciaBK committed:
Merge branch 'master' of https://github.com/AY2324S1-CS2113-T18-1/tp into tricia

TriciaBK committed:
Update PPP links in AboutUs

TongZhengHong committed:
Minor fixes

TriciaBK committed:
Update PPP

TongZhengHong committed:
Disable loadBoard method and remove saving

TriciaBK committed:
Add link to TP Dashboard

TongZhengHong committed:
Remove calls to loadBoard

TongZhengHong committed:
Fix checkstyle

TongZhengHong committed:
Fix checkstyle 2

TongZhengHong committed:
Merge pull request #199 from TriciaBK/tricia

Add PPP link in AboutUs

TongZhengHong committed:
Fix checkstyle 3

onx001 committed:
Merge pull request #200 from TongZhengHong/remove-chessboard-storag

Remove chessboard in storage

TongZhengHong committed:
Remove json tag in code block

ken-ruster committed:
Fix bug with current turn in save

onx001 committed:
Merge pull request #201 from TongZhengHong/branch-ug-format-4

Remove json tag in code block

ken-ruster committed:
Merge branch 'add-tests' of https://github.com/ken-ruster/tp into add-tests

onx001 committed:
Merge pull request #202 from ken-ruster/add-tests

Add tests

TongZhengHong committed:
Add gameplay link

ken-ruster committed:
Merge pull request #203 from TongZhengHong/branch-ug-format-4

Add gameplay link
